src/app/components/turnos/dar-turnos/dar-sobreturno.component.ts

```diff
@@ -0,0 +1,318 @@
+            if (resultado.length) {
+                this.obraSocialPaciente = resultado.map((os: any) => {
+                    let osPaciente;
+
+                    if (os.nombre) {
+                        osPaciente = {
+                            'id': os.nombre,
+                            'label': os.nombre
+                        };
+                    } else {
+                        osPaciente = {
+                            'id': os.financiador,
+                            'label': os.financiador
+                        };
+                    }
+                    return osPaciente;
+                });
+                this.modelo.obraSocial = this.obraSocialPaciente[0].label;
+            }
+            this.obraSocialPaciente.push({ 'id': 'prepaga', 'label': 'Prepaga' });
+        });
+    }
+
+    seleccionarObraSocial(event) {
+        if (event.value === 'prepaga') {
+            this.obraSocialService.getPrepagas().subscribe(prepagas => {
+                this.showListaPrepagas = true;
+                this.prepagas = prepagas;
+            });
+        } else {
+            this.showListaPrepagas = false;
+        }
+        this.modelo.obraSocial = event && event.value;
+    }
+
+    // Operaciones con carpetaPaciente
+
+    // Se busca el número de carpeta de la Historia Clínica en papel del paciente
+    // a partir del documento y del efector
+    obtenerCarpetaPaciente() {
+        let indiceCarpeta = -1;
+        if (this.paciente.carpetaEfectores.length) {
+            // Filtro por organizacion
+            indiceCarpeta = this.paciente.carpetaEfectores.findIndex(x => x.organizacion.id === this.auth.organizacion.id);
+            if (indiceCarpeta > -1) {
+                this.carpetaEfector = this.paciente.carpetaEfectores[indiceCarpeta];
+            }
+        }
+        if (indiceCarpeta === -1) {
+            // Si no hay carpeta en el paciente MPI, buscamos la carpeta en colección carpetaPaciente, usando el nro. de documento
+            this.serviceCarpetaPaciente.getNroCarpeta({ documento: this.paciente.documento, organizacion: this.auth.organizacion.id }).subscribe(carpeta => {
+                if (carpeta.nroCarpeta) {
+                    this.carpetaEfector.nroCarpeta = carpeta.nroCarpeta;
+                    this.changeCarpeta = true;
+                }
+            });
+        }
+    }
+
+    actualizarCarpetaPaciente() {
+        this.servicePaciente.patch(this.paciente.id, { op: 'updateCarpetaEfectores', carpetaEfectores: this.paciente.carpetaEfectores }).subscribe(resultadoCarpeta => {
+        });
+    }
+
+    // Se ejecuta al modificar el campo NroCarpeta
+    cambiarCarpeta() {
+        this.changeCarpeta = true;
+    }
+
+    verificarTelefono(paciente: IPaciente) {
+        // se busca entre los contactos si tiene un celular
+        this.telefono = '';
+        this.cambioTelefono = false;
+        if (paciente.contacto) {
+            if (paciente.contacto.length) {
```

Comment:
```suggestion
            if (paciente.contacto?.length) {
```
o incluso para la linea de abajo, sin el if anterior, tambien sería valido
```suggestion
            paciente.contacto?.forEach((contacto) => {
```